src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Utilities/StringExtensions.cs

```diff
@@ -361,23 +329,26 @@ public static string Translate(this string subject, char[] from, char[] to)
         public static string ReplaceAll(this string original, IDictionary<string, string> replacements)
         {
             var pattern = $"{string.Join("|", replacements.Keys)}";
+
             return Regex.Replace(original, pattern, match => replacements[match.Value]);
         }
 
-        public static string TrimEnd(this string rough, string trim = "")
+        public static string TrimEnd(this string source, string trimValue = "") => source.TrimEnd(trimValue.ToCharArray());
```

Comment:
source.TrimEnd(trimValue.ToCharArray()) this is different than `rough[..^trim.Length]`